### ISSUE TITLE
[feat]: Support configurable database options

### ISSUE THREAD

**Issue by Meisterzunge** [Start9Labs/lnbits-startos#46]
### Prerequisites

- [X] I have searched for existing issues that already suggest this feature.

### Describe the Feature Request

LNbits supports various database options (SQLite, Cockroach and Postgres).
When setting up the LNBits service in StartOS (currently v0.3.5.1) SQLite will be used as database. To quote the [LNBits docs](https://docs.lnbits.org/guide/installation.html#additional-guides), it is highly recommended to switch to postgres for running LNBits on scale.

### Describe the Use Case

Running LNBits on a SQLite database is fine if you want to play around. But you want to have a reliable and fast service when using it in a production environment.

### Describe Preferred Solution

I propose an option in the settings section to configure wich database to be used. No problem if this settings can only be changed before starting the service for the first time.

### Describe Alternatives

As an alternative I would recommend to use postgres instead of an SQLite database. No settings but using postgres as a one-fits-all solution.

**Comment by MattDHill** [comment]
Hi, are you experiencing performance issues with LNBits related to database operations? I would be shocked if this were the case. The amount of data being stored and the amount of connections is tiny, I don't see why Postgres would create any noticible performance or reliability benefit at that scale. Could you provide more information, perhaps I'm missing something

**Comment by Meisterzunge** [comment]
Hi,
I don't have any issues so far. But if postgres is that recommended like said in the LNBits docs, I don't feel comfortable running on a SQLite.

**Comment by Dominion5254** [comment]
Can you elaborate on why you "don't feel comfortable" using a SQLite DB? What specifically is your concern(s) here?

To quote the docs you linked above: "If you already have LNbits installed and running, on an SQLite database, we highly recommend you migrate to postgres *if you are planning to run LNbits on scale*" - that "if" seems quite important here. While I'm sure the LNbits team could provide better detail on the numbers, I suspect it would entail a very large number of users (far beyond family/friends) before any DB issues would present.

That said, this migration to Postgress will quite likely happen in the future, but it is unlikely to be a high priority.

**Comment by Meisterzunge** [comment]
I didn't want to get into conceivable database troubles.

For me the docs have been unambiguous. But I agree, there are some details on the numbers missing.
So I'm fine to see this feature on lower priority.

**Comment by MattDHill** [comment]
Closing this as not planned.

LNbits still treats SQLite as a first-class backend in 1.5.6: it is what `.env.example` configures by default, Postgres and CockroachDB sit beside it as alternatives, and nothing in the release marks SQLite as deprecated or on the way out.

The "at scale" wording in the docs you linked is the operative part. SQLite's single-writer lock is what gives way under many concurrent writers, and that is not the shape of a StartOS install — one server, one Lightning node behind it, an operator and the people they hand wallets to. Postgres would not make that measurably faster or more durable, and neither engine is what would fail first at that size.

The cost is not free, either. It means a Postgres subcontainer to run, health-check and back up alongside LNbits, plus a one-way SQLite-to-Postgres migration for every existing install — upstream's `tools/conv.py`, run against a database that holds balances. That is real failure surface bought for no gain anyone can point at.

If you or anyone else does hit an actual SQLite limit on StartOS — lock contention, a corrupted database, something with a symptom attached — please open a new issue with what you saw and we will look again. Thanks for raising it, and for the follow-up on priority.
